components/hab/src/command/pkg/uninstall_impl.rs

```diff
@@ -223,6 +225,22 @@ async fn supervisor_services() -> Result<Vec<PackageIdent>> {
     Ok(out)
 }
 
+#[derive(Clone, Copy)]
+enum UninstallIfRunning<'a> {
+    EvenIfRunning,
+    SkipIfRunning(&'a [PackageIdent]),
+}
```

Comment:
The naming of this is a little off, in that you can get statements like `UninstallIfLoaded::SkipIfLoaded(...)`, which seem contradictory.

Perhaps something like `UninstallMode::SkipIfLoaded`, etc.? Admittedly, I'm not in love with that name, but it feels like it's moving in a good direction (IMO, at least :sweat_smile: )

Reply:
That is pretty comical. I think it went through a few too many refactorings :sweat_smile: Addressed in [22b87b](https://github.com/habitat-sh/habitat/pull/7582/commits/22b87b3417cbd7943bea168a868cc1e2afa1c91b)